### PR TITLE
Remove ESM workarounds

### PR DESCRIPTION
**Pull request by alan-agius4**

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/webpack/configs/common.ts [hunk 1]
```diff
@@ -79,22 +79,11 @@ export async function getCommonConfig(wco: WebpackConfigOptions): Promise<Config
   const extraRules: RuleSetRule[] = [];
   const entryPoints: { [key: string]: [string, ...string[]] } = {};
 
-  // Load ESM `@angular/compiler-cli` using the TypeScript dynamic import workaround.
-  // Once TypeScript provides support for keeping the dynamic import this workaround can be
-  // changed to a direct dynamic import.
```
Comment (alfaproject) [inline packages/angular_devkit/build_angular/src/webpack/configs/common.ts:84]: This comment is still valid

File: packages/angular_devkit/build_angular/src/babel/webpack-loader.ts [hunk 2]
```diff
@@ -136,14 +136,11 @@ export default custom<ApplicationPresetOptions>(() => {
...
         // entry point exists.
-        // TODO_ESM: Make import failure an error once the `tools` entry point exists.
         if (i18nPluginCreators === undefined) {
           // Load ESM `@angular/localize/tools` using the TypeScript dynamic import workaround.
           // Once TypeScript provides support for keeping the dynamic import this workaround can be
           // changed to a direct dynamic import.
-          try {
-            i18nPluginCreators = await loadEsmModule<I18nPluginCreators>('@angular/localize/tools');
-          } catch {}
+          i18nPluginCreators = await loadEsmModule<I18nPluginCreators>('@angular/localize/tools');
```
Comment (dgp1130) [inline packages/angular_devkit/build_angular/src/babel/webpack-loader.ts:143]: Nit: Can we drop the if-statement now and simplify to `i18nPluginCreaters ??= await loadEsmModule(/* ... */);`?

**Review by dgp1130 — APPROVED** [review]
Thanks for cleaning this up! Glad we can remove this tech debt.

File: packages/angular_devkit/build_angular/src/babel/webpack-loader.ts (hunk 2 above)
Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/babel/webpack-loader.ts:143]: Good catch

**Pull request merged**